authorizer/src/utils/header_utils.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python
+# -*- coding: utf-8; -*-
+
+# Copyright (c) 2023 Oracle and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/
+import json
+from typing import Dict, List
+from urllib import request
+
+HEADERS_JSON_KEY = "data"
+AUTHORIZATION_KEY = "authorization"
+SIGNATURE_HEADERS_KEY = "headers"
+
+
+class MissingRequiredHeadersException(Exception):
+    def __init__(
+        self, 
+        required_headers: List[str], 
+        provided_headers: List[str]
+    ):
+        self.required_headers = required_headers
+        self.provided_headers = provided_headers
+
+    def __str__(self):
+        return "Headers required for authentication were not provided.\nProvided headers: {0}\nRequired headers: {1} \
+            \nMissing headers: {2}".format(self.provided_headers, self.required_headers, set(self.required_headers)
+                                           .difference(set(self.provided_headers)))
+
+
+class AuthorizationHeaderMissingException(Exception):
+    def __str__(self):
+        return "Expected authorization header to be present but was not found"
```

Comment:
NIT: Expected the authorization header to be present, but it was not found?

Reply:
Updated